### ISSUE TITLE
GoG Cloud Saves not being detected when syncing.

### ISSUE THREAD

**Issue by colinjmatt** [Heroic-Games-Launcher/HeroicGamesLauncher#3973]
### Describe the bug

When syncing cloud saves in Heroic to a GoG game (in my only available example, this is Fallout 4 GOTY), it doesn't detect any cloud saves, and (successfully) uploads any saves on the current machine every time a sync is attempted.

Uploads have been verified as successful through the GoG [cloud save management web UI](https://www.gog.com/account/cloud-saves) which I've tested multiple times by manually deleting all cloud saves from the web UI and attempting another sync. The result is any saves on the local machine successfully re-upload. When trying to sync on another local machine, it detects 0 files in the cloud and uploads any that are present on that local machine.

If there's a difference in save files (by filename only I think), all unique files are uploaded from both local machines.

I have attempted all 5 sync methods that I know of to try and download saves on 2 different machines:
- Via the Heroic UI to manually trigger one of the 4 sync options (Upload, Download, Force upload, and Force Download)
- Automatically when starting/exiting a game

A **potential** trigger for this may have been when I first manually cleared save files from the GoG web UI, as I have been able to upload AND download as recently as 6 days ago, which is about a day before I manually cleared them (there's a new 200mb save limit coming in from GoG). Since then, Heroic will not see any cloud files.

I have attempted a sync using GoG Galaxy 2.0 on a Windows machine, and that behaves as expected and pulls save down from GoG Cloud. I am experiencing no issues with Epic Cloud saves.

Logs attached here
[2024-08-21T19_35_43-gogdl.log](https://github.com/user-attachments/files/16716117/2024-08-21T19_35_43-gogdl.log)
[2024-08-21T19_35_43-heroic.log](https://github.com/user-attachments/files/16716119/2024-08-21T19_35_43-heroic.log)
[1998527297-lastPlay.log](https://github.com/user-attachments/files/16716120/1998527297-lastPlay.log)

### Add logs

```text
Logs attached in the description field.
```

### Steps to reproduce

Trigger the sync process in any way at all for a game that has saves stored in the GoG cloud.

### Expected behavior

Files should download/upload according to the sync action taken

### Heroic Version

Latest Stable

### System Information

- OS: `Arch Linux` and `SteamOS`
- Heroic install version: `Native` and `Flatpak` versions
- Prefixes are on `ext4` drives, 1 nVME (desktop) and 1 SD card (Steam Deck)
- RAM: yes
- CPU: yes
- Architecture: x86-64 AMD

**Comment by frpnit** [comment]
I can confirm the same happened to me and appears to be related with erasing save files from https://www.gog.com/account/cloud-saves ; i'm on windows so it's not OS specific issue

**Comment by imLinguin** [comment]
Indeed, there seems to be a corelation to that. I assume the server prevents from reading the cloud saves if there are files tagged for deletion **and the client doesn't specify it supports that**. We need to figure out how to specify that support and add respect files tagged for deletion

**Comment by auterak** [comment]
I think I've also encountered this problem with Cyberpunk 2077, where I manually removed some saves due to the new 200MB limit. It says that there are no files in the cloud when I try to sync the saves in Heroic.

I've tried another game (Chorus), where I did not remove any saves, and the cloud sync works just fine.

However I don't have any benchmarks if the cloud sync worked before, since I installed Linux and the games on my laptop after the fact. And on my main PC with Windows (source of the saves) it just works via Galaxy. I don't have other Windows machine to try it there.
